src/shared/telemetry/util.ts

```diff
@@ -90,6 +92,37 @@ export async function getUserAgent(
     return pairs.join(' ')
 }
 
+type EnvType =
+    | 'cloud9'
+    | 'cloud9-codecatalyst'
+    | 'amazon-cloud9'
+    | 'cloud9-ec2'
+    | 'codecatalyst'
+    | 'local'
+    | 'ec2'
+    | 'sagemaker'
+    | 'test'
+    | 'other'
+
+export function getComputeEnvType(): EnvType {
+    if (isCloud9('classic')) return 'cloud9'
+
+    if (isCloud9('codecatalyst')) return 'cloud9-codecatalyst'
+
+    if (isInDevEnv()) return 'codecatalyst'
+
+    if (isCn()) return 'amazon-cloud9'
```

Comment:
Is in China

Reply:
I'm not really sure. do u want to rename value to be?